src/Test/QuickCheck/Laws/Data/Foldable.purs

```diff
@@ -0,0 +1,57 @@
+module Test.QuickCheck.Laws.Data.Foldable where
+
+import Prelude
+import Control.Monad.Eff.Console (log)
+import Data.Foldable (foldMap, fold, foldlDefault, foldl, foldr, class Foldable, foldrDefault)
+import Test.QuickCheck (QC, quickCheck')
+import Test.QuickCheck.Arbitrary (class Arbitrary)
+import Test.QuickCheck.Laws (A, B)
+import Type.Proxy (Proxy2)
+
+
+-- | - foldr == foldrDefault
+-- | - foldl == foldlDefault
+-- | - fold == foldMap id
+checkFoldable
+  ∷ ∀ eff f
+  . (Foldable f, Arbitrary (f A))
+  ⇒ Proxy2 f
+  → QC eff Unit
+checkFoldable _ = do
+
+  log "Checking 'foldr' law for Foldable"
+  quickCheck' 1000 foldrLaw
+
+  log "Checking 'foldl' law for Foldable"
+  quickCheck' 1000 foldlLaw
+
+  log "Checking 'fold' law for Foldable"
+  quickCheck' 1000 foldLaw
+
+  where
+    foldrLaw :: (A -> B -> B) -> B -> f A -> Boolean
+    foldrLaw f z t = foldr f z t == foldrDefault f z t
+
+    foldlLaw :: (B -> A -> B) -> B -> f A -> Boolean
+    foldlLaw f z t = foldl f z t == foldlDefault f z t
+                                                           
+    foldLaw :: f A -> Boolean
+    foldLaw t = fold t == foldMap id t
```

Comment:
This is just the definition of `fold`, though: https://github.com/purescript/purescript-foldable-traversable/blob/17ddb52928dd00fdb31b1cde3931991d6ea9e366/src/Data/Foldable.purs#L166